app/main.py

```diff
@@ -1,8 +1,25 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.__class__.people[name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(people: dict) -> list:
+    for single_person in people:
+        Person(single_person["name"], single_person["age"])
+
+    for single_person in people:
+        person = Person.people[single_person["name"]]
+        spouse_name = single_person.get("wife") or single_person.get("husband")
+        if spouse_name:
+            spouse = Person.people[spouse_name]
+            if "wife" in single_person:
+                person.wife = spouse
+            else:
+                person.husband = spouse
+            person.spouse = spouse
```

Comment:
You can simply create 2 separeted checks via `get()` method . First check for wife and second fot husband . 


That way your code will be more clean